bap-vibes/src/congruence.ml

```diff
@@ -0,0 +1,15 @@
+open Bap.Std
+open Bap_core_theory
+
+type cls
+type t = cls KB.obj
+type computed = (cls, unit) KB.cls KB.value
+
+let package = "vibes"
+let name = "congruent-temps"
+let cls : (cls, unit) KB.cls = KB.Class.declare ~package name ()
+
+let slot : (cls, (var * var) option) KB.slot =
```

Comment:
Is not the congruence information a part of the patch? I would've expected something like a Set domain of var * var pairs in Patch.cls

Reply:
Like can we have collisions if you are compiling more than one patch (which we support?) ?

Reply:
This is true. Is there an easy way to query the KB for the current patch we're compiling? Otherwise we would have to thread it through the pipeline.

Reply:
Not that I know of. It may be possible. I have heard that the KB has a notion of changeable state now. It might be simpler and ultimately better to just thread though.

Reply:
We do already thread the patch up to `compile_one_vibes_ir`:
https://github.com/draperlaboratory/VIBES/blob/bmourad/unify-congruence/bap-vibes/src/compiler.ml#L79

We'd just need to thread it through to the Bir_passes and then Linear_ssa?

Reply:
That seems to be the most straightforward approach.